lib/shellable.ts

```diff
@@ -34,7 +45,7 @@ export interface ShellableOptions {
    *
    * @default No additional environment variables
    */
-  environment?: { [key: string]: string };
+  environment?: { [key: string]: string | ShellableEnvironmentVariable };
```

Comment:
This is not jsii-able (although we don’t build this module with jsii currently). What you could do is just add another field “environmentSecrets” which will allow specifying secret ARNs for values. 

But I am also ok with this.

Reply:
yeah wondering what the possibility of building with JSII is for delivlib. Was just looking for the best way to add this without breaking the api.

I see what you mean with `environmentSecrets`. We would have to do the same for ssm parameters with `environmentParameters` or something similar. I'll explore a bit but first instinct was to mirror the codebuild construct API.

Reply:
yeah wondering what the possibility of building with JSII is for delivlib. Was just looking for the best way to add this without breaking the api.

I see what you mean with `environmentSecrets`. We would have to do the same for ssm parameters with `environmentParameters` or something similar. I'll explore a bit but first instinct was to mirror the codebuild construct API.

Reply:
yeah wondering what the possibility of building with JSII is for delivlib. Was just looking for the best way to add this without breaking the api.

I see what you mean with `environmentSecrets`. We would have to do the same for ssm parameters with `environmentParameters` or something similar. I'll explore a bit but first instinct was to mirror the codebuild construct API.